inbm/dispatcher-agent/dispatcher/ota_parser.py

```diff
@@ -176,8 +176,9 @@ def parse(self, resource: Dict, kwargs: Dict, parsed: XmlHandler) -> Dict[str, A
             logger.info("Config Params not passed correctly"
                         " in manifest, rejected update")
             raise XmlException
-        kwargs.update({'signature': self._signature, 'config_params': config_params,
+        kwargs.update({'config_params': config_params,
                        'hash_algorithm': self._hash_algorithm,
+                       'signature': self._signature,
                        'app_type': app,
                        'cmd': cmd,
                        'container_tag': container_tag,
```

Comment:
**Suggestion:** Ensure that the 'signature' key is only updated in 'kwargs' if it is present in the 'resource' dictionary. This avoids overwriting with 'None' if 'signature' is not provided, which could lead to unintended behavior. [enhancement]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
        if self._signature:
            kwargs.update({'signature': self._signature})
        kwargs.update({'config_params': config_params,
                       'hash_algorithm': self._hash_algorithm,
                       'app_type': app,
                       'cmd': cmd,
                       'container_tag': container_tag})
```